include/arch/zx/sprites/sp1.h

```diff
@@ -217,6 +217,7 @@ extern void               __LIB__   sp1_MoveSprAbs(struct sp1_ss *s, struct sp1_
 extern void               __LIB__   sp1_MoveSprRel(struct sp1_ss *s, struct sp1_Rect *clip, void *frame, char rel_row, char rel_col, char rel_vrot, char rel_hrot);
 extern void               __LIB__   sp1_MoveSprPix(struct sp1_ss *s, struct sp1_Rect *clip, void *frame, uint16_t x, uint16_t y);
 
+// hook1 and hook2 must have prototype: void userfunc( uint8_t count, struct sp1_cs *c ) __smallc;
```

Comment:
If this the case then prototype properly - if there's parsing issues then a `typedef void (*sp1_hook_func)( uint8_t count, struct sp1_cs *c ) __smallc` should get around them.

Reply:
The function can have __z88dk_fastcall or not, and the prototype would be different for each. The functions using the hooks can use both types, so in order to be compatible, a `void *` pointer needs to be passed.